### DISCUSSION TITLE
How to change password

### DISCUSSION THREAD

**Discussion by mariobrosch** [petretiandrea/home-assistant-tapo-p100#327] [Q&A] [answered]
Since my password was unsafe, I changed it in my tp-link account, but how can I change it in home assistant for this controller?

**Comment by busywait** [comment]
Hey @mariobrosch, did you find the answer for this? I'm sure that it's simple to do but I can't find any UI or file to edit :/

**Comment by mariobrosch** [comment]
Hi @busywait , no I did not find a solution without removing the specific controller and re-add it.

**Reply by busywait** [reply]
Thanks - yes I've seen notes in [#122 Invalid authentication](https://github.com/petretiandrea/home-assistant-tapo-p100/issues/122) that suggest removing and re-adding devices in both the tapo app and Home Assistant. I've taken the route of changing my password back to what it was before - then devices started working again :/

**Comment by petretiandrea** [comment, marked as answer]
Actually, you need to remove and re-add the device by using new password
